CodingTracker.AlvinGraham/CodingTracker.AlvinGraham/UserInterface.cs

```diff
@@ -0,0 +1,183 @@
+						LiveTrack();
+						break;
+					case MainMenuChoices.FilterRecords:
+						FilterRecords();
+						break;
+					case MainMenuChoices.Reports:
+						Utilities.ClearScreen("Reports Menu");
+						RunReports();
+						break;
+					case MainMenuChoices.ManageGoals:
+						Utilities.ClearScreen("Managing Goals");
+						ManageGoals();
+						break;
+					case MainMenuChoices.Quit:
+						Console.WriteLine("Goodbye");
+						isMenuRunning = false;
+						break;
+				}
+			}
+			catch (ArgumentException)
+			{
+				Console.WriteLine("\nReturning to Main Menu.  Press any key to continue...");
+				Console.ReadKey();
+			}
+		}
+	}
+
+	private static void ManageGoals()
+	{
+		var goalsSession = new Goals();
+		goalsSession.GoalsMenu();
+	}
+
+	private static void RunReports()
+	{
+		var reportSession = new Reports();
+		reportSession.RunReports();
+		Console.WriteLine("\nPress any Key to Return to main menu.");
+		Console.ReadKey();
+	}
+
+	private static void FilterRecords()
+	{
+		var filterSession = new RecordFilter();
+		filterSession.FilterRecordsMenu();
+		Console.WriteLine("\nPress any Key to Return to main menu.");
+		Console.ReadKey();
+	}
+
+	private static void LiveTrack()
+	{
+		var trackingSession = new LiveTracker();
+		trackingSession.TrackSession();
+		Console.WriteLine("\nPress any Key to Return to main menu.");
+		Console.ReadKey();
+	}
+
+	private static void DeleteRecord()
+	{
+		var dataAccess = new DataAccess();
+		var records = dataAccess.GetAllRecords();
+		ViewRecords(records);
+
+		var id = Utilities.GetNumber("Please type the id of the habit you want to delete.");
+
+		if (!AnsiConsole.Confirm("Are you sure?"))
+			return;
+
+		var response = dataAccess.DeleteRecord(id);
+
+		var responseMessage = response < 1
+			? $"Record with id {id} doesn't exit. Press any key to return to Main Menu."
+			: "Record deleted successfully. Press any key to return to Main Menu";
+
+		Console.WriteLine(responseMessage);
+		Console.ReadKey();
+	}
+
+	private static void UpdateRecord()
+	{
+		var dataAccess = new DataAccess();
+		var records = dataAccess.GetAllRecords();
+		ViewRecords(records);
+
+		var id = Utilities.GetNumber("Please type the id of the habit you want to update.");
+
+		var record = records.Where(x => x.Id == id).Single();
+		var dates = Utilities.GetDateInputs();
+
+		record.DateStart = dates[0];
+		record.DateEnd = dates[1];
+
+		dataAccess.UpdateRecord(record);
+	}
+
+	private static void ViewRecords(IEnumerable<CodingRecord> records)
+	{
+		var table = new Table();
+		table.AddColumn("Id");
+		table.AddColumn("Start Date");
+		table.AddColumn("End Date");
+		table.AddColumn("Duration");
+
+		foreach (var record in records)
+		{
+			table.AddRow(record.Id.ToString(), record.DateStart.ToString(), record.DateEnd.ToString(), $"{(int)record.Duration.TotalHours} hours {record.Duration.TotalMinutes % 60} minutes");
+		}
+
+		AnsiConsole.Write(table);
+
+		Console.WriteLine("Press any key to return to main menu.");
```

Comment:
When viewing the Updating/Deleting records, kindly removed the “Press any key to return to main menu.”
When user enter 0, we are back main menu. It will be better to update the messages.
Please type the id of the habit you want to update.

Reply:
Ha, I inadvertently fixed this while resolving issue #2 above!